homeassistant/components/autarco/sensor.py

```diff
@@ -117,9 +188,52 @@ async def async_setup_entry(
             for description in SENSORS_INVERTER
             for inverter in coordinator.data.inverters
         )
+        if coordinator.data.battery:
+            entities.extend(
+                AutarcoBatterySensorEntity(
+                    coordinator=coordinator,
+                    description=description,
+                )
+                for description in SENSORS_BATTERY
+            )
     async_add_entities(entities)
 
 
+class AutarcoBatterySensorEntity(
+    CoordinatorEntity[AutarcoDataUpdateCoordinator], SensorEntity
+):
+    """Defines an Autarco battery sensor."""
+
+    entity_description: AutarcoBatterySensorEntityDescription
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        *,
+        coordinator: AutarcoDataUpdateCoordinator,
+        description: AutarcoBatterySensorEntityDescription,
+    ) -> None:
+        """Initialize Autarco sensor."""
+        super().__init__(coordinator)
+
+        self.entity_description = description
+        self._attr_unique_id = (
+            f"{coordinator.account_site.site_id}_battery_{description.key}"
+        )
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, f"{coordinator.account_site.site_id}_battery")},
+            entry_type=DeviceEntryType.SERVICE,
+            manufacturer="Autarco",
+            name="Battery",
+        )
```

Comment:
Should we add `via_device`?

Reply:
You cannot link this to another device, as a battery is a separate entity within the Autarco ecosystem.

Reply:
But it's a physical thing that lives in your home and that's a device tbh

Reply:
A device from which we do not retrieve specific device information with the API... For me it is only a device if I can also attach a unique ID to it.

Autarco does not treat it as a device like they do with the inverters. Suppose someone had multiple batteries, then we still have to deal with summed values ​​from the Autarco API.